biz.aQute.resolve/src/biz/aQute/resolve/Bndrun.java

```diff
@@ -157,4 +159,54 @@ public int compare(VersionedClause a, VersionedClause b) {
 			}
 		}
 	}
+
+	class Node {
+		final Resource	r;
+		final Set<Node>	children	= new HashSet<>();
+
+		Node(Resource resource) {
+			r = resource;
+		}
+
+		public void addDependents(List<Node> resources) {
+			if (resources.contains(this))
+				return;
+
+			for (Node c : children) {
+				c.addDependents(resources);
+			}
+
+			// if we have cycles, break them
+			if (resources.contains(this))
+				return;
+
+			resources.add(this);
+		}
+	}
+
+	public Collection<Resource> getResources(Map<Resource, List<Wire>> resolution) {
```

Comment:
Perhaps this method should not be `public` to avoid creating API where we don't really need the method to be API?